lib/connection.js

```diff
@@ -1380,11 +1381,37 @@ Connection.prototype.setClient = function setClient(client) {
   return this;
 };
 
-Connection.prototype.syncIndexes = async function syncIndexes() {
+/**
+ *
+ * Syncs all the indexes for the models registered with this connection.
+ *
+ * @param {Object} options
+ * @param {Boolean} options.continueOnError `false` by default. If set to `true`, mongoose will not throw an error if one model syncing failed, and will return an object where the keys are the names of the models, and the values are the results/errors for each model.
+ * @returns
+ */
+Connection.prototype.syncIndexes = async function syncIndexes({ continueOnError } = { continueOnError: false }) {
   const result = {};
-  for (const model in this.models) {
-    result[model] = await this.model(model).syncIndexes();
+  const errorsMap = { };
+
+  for (const model of Object.values(this.models)) {
+    try {
+      result[model.modelName] = await model.syncIndexes();
+    } catch (err) {
+      if (!continueOnError) {
+        errorsMap[model.modelName] = err;
+        break;
+      } else {
+        result[model.modelName] = err;
+      }
+    }
   }
+
+  if (!continueOnError && Object.keys(errorsMap).length) {
+    const message = Object.values(errorsMap).map(err => err.message).join(' ');
```

Comment:
What do you think about making this formatting consistent with CastErrors, so we do `${key}: ${err.message}` and join with `,`?